### PR TITLE
Penalty disbursal address

### PR DESCRIPTION
**Pull request by rudolfix**

### CODE REVIEW

File: contracts/ApproveAndCallCallback.sol [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pragma solidity 0.4.15;
+
+contract ApproveAndCallFallBack {
```
Comment (krzkaczor) [inline contracts/ApproveAndCallCallback.sol]: make it an interface perhaps?

File: test/LockedAccount.js [hunk 2]
```diff
@@ -6,10 +6,13 @@ import eventValue from "./helpers/eventValue";
 import * as chain from "./helpers/spawnContracts";
 import increaseTime, { setTimeTo } from "./helpers/increaseTime";
 import latestTime, { latestTimestamp } from "./helpers/latestTime";
+import EVMThrow from "./helpers/EVMThrow";
 
-const LockedAccount = artifacts.require("LockedAccount");
+const TestFeeDistributionPool = artifacts.require("TestFeeDistributionPool");
 
-contract("LockedAccount", accounts => {
+const should = require("chai").should();
```
Comment (krzkaczor) [inline test/LockedAccount.js]: Just modify  1st line `import { expect, should } from "chai";`

File: test/LockedAccount.js [hunk 3]
```diff
@@ -29,119 +32,242 @@ contract("LockedAccount", accounts => {
...
-      value: ticket / 2,
-      from: investor2,
-    });
+    // verify longstop date independently, value is convertable to int so do it
+    let unlockDate = timebase + 18 * 30 * chain.days;
+    if (parseInt(initialLockedBalance[2]) > 0) {
+      // earliest date is preserved for repeated investor address
+      unlockDate = parseInt(initialLockedBalance[2]);
+    }
+    assert.equal(parseInt(investorBalance[2]), unlockDate, "18 months in future");
```
Comment (krzkaczor) [inline test/LockedAccount.js]: use expect

File: contracts/EtherToken.sol [hunk 4]
```diff
@@ -14,6 +15,21 @@ contract EtherToken is StandardToken, TokenWithDeposit, Math {
     // disable default function
     function() { revert(); }
 
+    function approveAndCall(address _spender, uint256 _amount, bytes _extraData)
```
Comment (krzkaczor) [inline contracts/EtherToken.sol]: shouldn't this be part of contract `ERC23`? This seems like quite generic implementation

File: contracts/LockedAccount.sol [hunk 5]
```diff
@@ -95,57 +95,71 @@ contract LockedAccount is Ownable, TimeSource, ReturnsErrors, Math {
...
+        onlyStates(LockState.AcceptingUnlocks, LockState.ReleaseAll)
+        public
+        returns (Status)
+    {
+        return unlockFor(msg.sender);
+    }
+
     // this allows to unlock and allow neumarks to be burned in one transaction
     function receiveApproval(address from, uint256 _amount, address _token, bytes _data)
+        onlyStates(LockState.AcceptingUnlocks, LockState.ReleaseAll)
```
Comment (krzkaczor) [inline contracts/LockedAccount.sol]: what if its called not by a token contract?

File: contracts/EtherToken.sol (hunk 4 above)
Comment (rudolfix) [inline contracts/EtherToken.sol]: ERC23 is interface and should stay like that. I hope we can fix truffle

File: contracts/LockedAccount.sol (hunk 5 above)
Comment (rudolfix) [inline contracts/LockedAccount.sol]: then you have this: `require(_token == address(neumarkToken));` check below. but a good catch! I have a test for this case as well

File: contracts/ERC23.sol [hunk 6]
```diff
@@ -0,0 +1,7 @@
+pragma solidity ^0.4.11;
```
Comment (rudolfix) [inline contracts/ERC23.sol]: Is it still a problem? I think a separate PR with just this change should be better

File: contracts/ApproveAndCallCallback.sol (hunk 1 above)
Comment (rudolfix) [inline contracts/ApproveAndCallCallback.sol]: I copied this from remco's minimetoken.... We'll align it separaterly OK?

Comment (rudolfix) [inline contracts/ApproveAndCallCallback.sol]: it was, then I bailed out on truffle problems

File: contracts/LockedAccount.sol (hunk 5 above)
Comment (krzkaczor) [inline contracts/LockedAccount.sol]: i think we should more modifiers in a code, generally speaking

**Pull request merged**